CHANGELOG.md

```diff
@@ -22,7 +22,7 @@
 
 ## Improvements
 
--
+- [core] Uses `@sap/xssec` for requesting `user token` and `client credentials token` from the XSUAA service instead of our own implementations.
```

Comment:
For me this is an implementation detail. Does this affect the user in any way? (Like - is the behavior any different maybe?) Otherwise, I think it shouldn't be in the changelog. What do you think?

Reply:
I mentioned this just because this would be a note for both users and us about the potential unexpected behaviour after migrating to a new lib. I feel the migration might be risky even we have e2e tests.

Reply:
But then maybe we mention it in the compatibility notes?